totp.go

```diff
@@ -0,0 +1,72 @@
+package walletclient
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/bitcoin-sv/spv-wallet-go-client/totp"
+	"github.com/bitcoin-sv/spv-wallet-go-client/utils"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/bitcoinschema/go-bitcoin/v2"
+	"github.com/libsv/go-bk/bip32"
+)
+
+// GenerateTotpForContact creates one time-based one-time password based on secret shared between the user and the contact
+func (b *WalletClient) GenerateTotpForContact(contact *models.Contact, validationPeriod, digits uint) (string, error) {
+	if b.xPriv == nil {
+		return "", errors.New("init client with xPriv first")
+	}
+
+	xpriv, err := deriveXprivForPki(b.xPriv)
+	if err != nil {
+		return "", err
+	}
+
+	cXpub, err := bip32.NewKeyFromString(contact.PubKey)
+	if err != nil {
+		return "", fmt.Errorf("contact's PubKey is invalid: %w", err)
+	}
```

Comment:
Why not make this a part of `sharedSecret(...)` method?

Reply:
To make totp more flexible and independent from contacts

Reply:
file removed